### ISSUE TITLE
Inline Initialized Fields Are Not Considered Nullable Before Initialization

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#408]
```
The following code will produce a NPE at runtime, but does not cause an error 
in the checker framework:

/*>>>import 
org.checkerframework.checker.initialization.qual.UnderInitialization;*/

class Foo {
  static class Bar {
    Bar() {
      doFoo();
    }

    String doFoo(/*>>>@UnderInitialization Bar this*/) {
      return "";
    }
  }

  static class Baz extends Bar {
    String myString = "hello";

    @Override
    String doFoo(/*>>>@UnderInitialization Baz this*/) {
      return myString.toLowerCase();
    }
  }

  public static void main(String[] args) {
    new Baz();
  }
}

Moving the initialization of myString into an anonymous initializer correctly 
causes the checker to emit an error:

/*>>>import 
org.checkerframework.checker.initialization.qual.UnderInitialization;*/

class Foo {
  static class Bar {
    Bar() {
      doFoo();
    }

    String doFoo(/*>>>@UnderInitialization(Bar.class) Bar this*/) {
      return "";
    }
  }

  static class Baz extends Bar {
    String hello;

    {
      hello = "world";
    }

    @Override
    String doFoo(/*>>>@UnderInitialization(Bar.class) Baz this*/) {
      return hello.toLowerCase();
    }
  }

  public static void main(String[] args) {
    new Baz();
  }
}
```

Original issue reported on code.google.com by `eatnumb...@google.com` on 9 Mar 2015 at 8:37

**Comment by smillst** [comment]
#779 was a duplicate of this issue.  I added this test case, checker/tests/nullness/Issue408.java and it passes.
